src/cclib/io/ccio.py

```diff
@@ -345,39 +354,37 @@ def _determine_output_format(outputtype, outputdest):
 
     Inputs:
       outputtype - a string corresponding to the file type
-        (one of cjson/json, cml, xyz)
       outputdest - a filename string or file handle
     Returns:
       outputclass - the class corresponding to the correct output format
+    Raises:
+      KeyError for unsupported file writer extensions
     """
 
     # Priority for determining the correct output format:
     #  1. outputtype
     #  2. outputdest
 
+    outputclass = None
     # First check outputtype.
     if isinstance(outputtype, str):
-        if outputtype.lower() in ('cjson', 'json'):
-            outputclass = cjsonwriter.CJSON
-        elif outputtype.lower() == 'cml':
-            outputclass = cmlwriter.CML
-        elif outputtype.lower() == 'xyz':
-            outputclass = xyzwriter.XYZ
+        try:
+            outputclass = outputclasses[outputtype.lower()]
+        except KeyError as e:
+            e.args = ('Unsupported Extension %s' % e.args[0],)
+            raise
```

Comment:
This part has been refactored to check if the extension is supported; and if not, raise a custom exception `UnknownOutputFormatError`.